dotCMS/src/main/java/com/dotcms/rendering/js/RequestBodyJavascriptReader.java

```diff
@@ -0,0 +1,37 @@
+package com.dotcms.rendering.js;
+
+import com.dotcms.rendering.util.ScriptingReaderParams;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.RoleAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.util.Logger;
+
+import java.io.Reader;
+import java.io.StringReader;
+
+/**
+ * This strategy reads the javascript code from a {@link java.util.Map} in the entry mapped to a convention-based
+ * key {@link RequestBodyJavascriptReader#EMBEDDED_JAVASCRIPT_KEY_NAME}
+ */
+
+public class RequestBodyJavascriptReader implements JavascriptReader {
+
+    static final String EMBEDDED_JAVASCRIPT_KEY_NAME = "javascript";
+
+    private static final String SCRIPTING_USER_ROLE_KEY = "Scripting Developer";
+
+    @Override
+    public Reader getJavaScriptReader(final ScriptingReaderParams params) throws DotSecurityException, DotDataException {
+
+        final RoleAPI roleAPI = APILocator.getRoleAPI();
+        final boolean canRenderVelocity = APILocator.getRoleAPI()
+                .doesUserHaveRole(params.getUser(), roleAPI.loadRoleByKey(SCRIPTING_USER_ROLE_KEY));
```

Comment:
So, to execute js you need scripting permissions?  Even for the API?  The way the permissions should work is that they should be allowed to execute JS if the last person to edit the `.js` file that is being executed had SCRIPTING_USER.

Reply:
This work on the exact same way that the VTL API works
Keep in mind this is the one that handles the dynamic endpoint